shop/shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,63 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.exception.NoSuchCategoryExistsException;
+import ro.msg.learning.shop.exception.NoSuchSupplierExistsException;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.SupplierRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService implements IProductService{
+    @Autowired
```

Comment:
Try to use the trick that I mentioned here too. Make them all `private final` and use `@RequiredArgsConstructor`